testsupport/tiers/nstemplateset.go

```diff
@@ -142,6 +143,42 @@ func UntilNSTemplateSetHasTemplateRefs(expectedRevisions TemplateRefs) wait.NSTe
 	}
 }
 
+// UntilNSTemplateSetHasStatusTemplateRefs checks if the NSTemplateTier has the expected template refs in the Status
+func UntilNSTemplateSetHasStatusTemplateRefs(expectedRevisions TemplateRefs) wait.NSTemplateSetWaitCriterion {
+	return wait.NSTemplateSetWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.NSTemplateSet) bool {
+			if expectedRevisions.ClusterResources == nil ||
+				actual.Status.ClusterResources == nil ||
+				*expectedRevisions.ClusterResources != actual.Status.ClusterResources.TemplateRef {
```

Comment:
shouldn't this rather be:
```suggestion
			if (expectedRevisions.ClusterResources == nil && actual.Status.ClusterResources != nil) ||
				(expectedRevisions.ClusterResources != nil && actual.Status.ClusterResources == nil) ||
				*expectedRevisions.ClusterResources != actual.Status.ClusterResources.TemplateRef {
```
or is there some small detail that I'm missing?

Reply:
btw, couldn't we simply verify that the status equals the spec? checking that the spec matches the expected revisions is a separate verification.

Reply:
> shouldn't this rather be

I've copied the logic from `UntilNSTemplateSetHasTemplateRefs` without looking very carefully into it 🤦‍♂️ . I've fixed that now in https://github.com/codeready-toolchain/toolchain-e2e/pull/1137/commits/db0fa2bcc320561a41b48289f421609c9d3f96ea

Reply:
> btw, couldn't we simply verify that the status equals the spec? checking that the spec matches the expected revisions is a separate verification.

Yeah, initially I wanted to check for the expected templaterefs from the NSTemplateTier in order to decouple the logic from the spec checks, just in case there's an issue with that and we set wrong/unwanted templateref there. But I agree that checking that comparing the spec and status refs should be enough here. Done in https://github.com/codeready-toolchain/toolchain-e2e/commit/db0fa2bcc320561a41b48289f421609c9d3f96ea